pkg/controller/master/virtualmachine/vmi_controller.go

```diff
@@ -143,3 +148,27 @@ func (h *VMIController) ReconcileFromHostLabels(_ string, vmi *kubevirtv1.Virtua
 
 	return vmi, nil
 }
+
+func (h *VMIController) SetHaltIfOccurExceededQuota(_ string, vmi *kubevirtv1.VirtualMachineInstance) (*kubevirtv1.VirtualMachineInstance, error) {
+	if vmi == nil || vmi.DeletionTimestamp != nil || vmi.Status.Conditions == nil {
+		return vmi, nil
+	}
+
+	for _, condition := range vmi.Status.Conditions {
+		if condition.Type == kubevirtv1.VirtualMachineInstanceSynchronized &&
+			condition.Status == corev1.ConditionFalse &&
+			strings.Contains(condition.Message, "exceeded quota") {
+
+			vm, err := h.virtualMachineCache.Get(vmi.Namespace, vmi.Name)
+			if err != nil {
+				return vmi, fmt.Errorf("failed to get VirtualMachine referred by VirtualMachineInstance(%s/%s): %w", vmi.Namespace, vmi.Name, err)
```

Comment:
I think the error is quite obvious and you should be able to return it directly.